internal/handlers/thread/template.go

```diff
@@ -88,31 +89,73 @@ func (h *TemplateHandler) ListPostsForThread(c echo.Context) error {
 		return err
 	}
 
-	err = c.Render(200, "posts", posts)
+	return c.Render(200, "posts", posts)
+}
+
+func (h *TemplateHandler) Post(c echo.Context) error {
+	postID := c.Param("id")
+	idAsInt, err := strconv.Atoi(postID)
 	if err != nil {
 		c.String(500, err.Error())
 		return err
 	}
 
-	return nil
+	post, err := h.threadService.GetPostByID(idAsInt)
+	if err != nil {
+		c.String(500, err.Error())
+		return err
+	}
+
+	return c.Render(200, "post", post)
 }
 
-func (h *TemplateHandler) Post(c echo.Context) error {
-	postID := c.Param("id")
+func (h *TemplateHandler) ThreadReply(c echo.Context) error {
+	values, err := c.FormParams()
+	if err != nil {
+		c.String(500, err.Error())
+		return err
+	}
 
-	idAsInt, err := strconv.Atoi(postID)
+	threadID := values.Get("thread_id")
+	threadIDAsInt, err := strconv.Atoi(threadID)
 	if err != nil {
 		c.String(500, err.Error())
 		return err
 	}
 
-	post, err := h.threadService.GetPostByID(idAsInt)
+	author := values.Get("member_name")
+	body := values.Get("body")
+
+	ip := c.RealIP()
+
+	postID, err := h.threadService.NewPost(body, ip, author, threadIDAsInt)
 	if err != nil {
 		c.String(500, err.Error())
 		return err
 	}
 
-	return c.Render(200, "post", post)
+	return c.JSON(200, NewPostResponse{
+		PostID: postID,
+	})
+}
+
+func (h *TemplateHandler) CreateThread(c echo.Context) error {
+	body := c.FormValue("body")
+	subject := c.FormValue("subject")
+
+	ip := c.RealIP()
+
+	author := c.FormValue("member")
+
+	threadID, err := h.threadService.NewThread(author, ip, body, subject)
+	if err != nil {
+		c.String(500, err.Error())
+		return err
+	}
+
+	return c.JSON(200, NewThreadResponse{
```

Comment:
same comment. let's leave template rendering in this file and put json stuff in the http file